### PR TITLE
Recap lab: Fix exercises and publish solution 

### PR DESCRIPTION
**Pull request by teodutu**

### CODE REVIEW

File: labs/recap/skel/09-syscall/pid_test.c [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/**
```
Comment (sketch873) [inline labs/recap/skel/09-syscall/pid_test.c:1]: Please remove extra tabs/spaces

File: labs/recap/skel/10-fork/fork.c [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
 #include <sys/types.h>
-
+#include <sys/wait.h>
 
 int main(void)
 {
 	int ii = 0;
 	
-	if (fork() >  0) ii++;
+	if (fork() >  0)
```
Comment (sketch873) [inline labs/recap/skel/10-fork/fork.c:17]: Please remove extra tabs

File: labs/recap/skel/11-fds/fd.c [hunk 3]
```diff
@@ -6,66 +13,69 @@
 
 #define MAGIC 17
 
-void do_work(int i);
+static void do_work(int i)
+{
```
Comment (sketch873) [inline labs/recap/skel/11-fds/fd.c]: Please remove extra tab

File: labs/recap/skel/11-fds/fd.c [hunk 4]
```diff
@@ -6,66 +13,69 @@
...
+static void do_work(int i)
+{	
+	int p[2];
+	int rd;
+	int wr;
+	int status;
+	int bytes;
+	pid_t pid;
+	char buf[10];
+
```
Comment (sketch873) [inline labs/recap/skel/11-fds/fd.c]: Another tab here

File: labs/recap/skel/12-file/file.c [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
  */
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <fcntl.h>
-#include <unistd.h>
-	
-	
+#include <unistd.h>
```
Comment (sketch873) [inline labs/recap/skel/12-file/file.c]: Please remove extra tab

Comment (sketch873) [reply labs/recap/skel/12-file/file.c]: Actually there are two, the second one on line 20.

**Review by sketch873 — CHANGES_REQUESTED** [review]
It looks nice. There are only few suggestions regarding extra tabs/spaces

**Review by lucigrigo — DISMISSED** [review]

**Review by sketch873 — DISMISSED** [review]

File: labs/recap/skel/01-cow/cow.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+			millis_end - millis_start);
+
+		break;
+
+	default:	/* Parent process */
+		rc = waitpid(pid, NULL, 0);
+		DIE(rc < 0, "waitpid");
+		break;
+	}
+
```
Comment (andreia-oca) [inline labs/recap/skel/01-cow/cow.c:72]: Please remove this extra empty line.

File: labs/recap/sol/02-io/README.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# 02: IO
+
+Apelul [`splice`](https://man7.org/linux/man-pages/man2/splice.2.html) implementează un mecanism de tipy `ZeroCopy`; el copiază dintr-un pipe într-un file descriptor.
+
+Datorită existenţei unui cache în sistemul de operare în care sunt menţinute rezultatele operaţiilor IO **(pagecache)**, o a doua accesare a fişierului se va face mult mai rapid, datele servindu-se din acest cache.
+
+La trunchierea fişierului se lucrează cu o structură ce este menţinută în memorie **(inode)**. Deci diferenţele nu vor fi foarte mari.
+
+Dacă folosim un fişier care este deja în memorie (cum ar fi `/dev/zero`), se va vedea o diferenţă notabilă doar la trunchierea fişierului.
```
Comment (andreia-oca) [inline labs/recap/sol/02-io/README.md]: IMO This should be translated into English.

File: labs/recap/sol/02-io/run_out [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+0:06.76
+
+Suport fizic:   1       In cache:       0       Exista output:  1
+0:05.11
+
+Suport fizic:   1       In cache:       1       Exista output:  0
+0:04.76
+
+Suport fizic:   1       In cache:       1       Exista output:  1
+0:04.19
```
Comment (andreia-oca) [inline labs/recap/sol/02-io/run_out]: Same as above, I think this note should be translated in English.

File: labs/recap/sol/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+Paginile sunt mapate read-only (copy-on-write) și sunt accesate doar pentru citire.
+Nu este alterat conținutul și nu apar page faulturi.
+
+* După al doilea ciclu `for` din procesul copil apar `NUM_PAGES / 2` faulturi.
+Paginile sunt mapate read-only, iar fiecare acces va genera un page fault (copy-on-write) și maparea paginii în forma read-write.
+
+* După încheierea procesului copil, în procesul părinte sunt vizibile page faulturile apărute.
+
+* În urma ultimului ciclu `for` (procesul părinte), se generează `NUM_PAGES` page faulturi.
+Paginile mapate inițial read-only sunt remapate read-write în urma page faulturilor.
```
Comment (andreia-oca) [inline labs/recap/sol/README.md]: Same as above, I think this note should be translated in English.

**Review by andreia-oca — CHANGES_REQUESTED** [review]
Looks great to me.

There are some notes in `labs/recap/sol/` that are currently in Romanian:
 - `labs/recap/sol/README.md`
 - `labs/recap/sol/02-io/run_out`
 - `labs/recap/sol/02-io/README.md`

IMO, I think we should translate them in English.

**Review by andreia-oca — APPROVED** [review]
Thanks. Looks great to me.

**Pull request merged**